portal/component-tests/partials/primary-navigation.component-test.js

```diff
@@ -1,50 +1,86 @@
-const { NON_MAKER_OR_CHECKER_OR_ADMIN_ROLES } = require('../../test-helpers/common-role-lists');
+const ROLES = require('../../server/constants/roles');
 const pageRenderer = require('../pageRenderer');
 
 const page = '_partials/primary-navigation.njk';
 const render = pageRenderer(page);
 
-const rolesToDisplayAllNavigationItems = [ADMIN];
-const rolesToDisplayHomeAndReportsNavigationItems = [MAKER, CHECKER];
-const rolesToDisplayHomeNavigationItem = NON_MAKER_OR_CHECKER_OR_ADMIN_ROLES;
-
 describe(page, () => {
   let wrapper;
 
-  describe.each(rolesToDisplayAllNavigationItems)('viewed by a %s', (role) => {
-    const user = { roles: [role] };
-
+  describe(`viewed by role '${ROLES.MAKER}'`, () => {
     beforeAll(() => {
-      wrapper = render({ user });
+      wrapper = render({ user: { roles: [ROLES.MAKER] } });
     });
 
     itRendersAHomeLink();
     itRendersAReportsLink();
-    itRendersAUsersLink();
-  });
 
-  describe.each(rolesToDisplayHomeAndReportsNavigationItems)('viewed by a %s', (role) => {
-    const user = { roles: [role] };
+    itDoesNotRenderAUtilisationReportUploadLink();
+    itDoesNotRenderAPreviousReportsLink();
+    itDoesNotRenderAUsersLink();
+  });
 
+  describe(`viewed by role '${ROLES.CHECKER}'`, () => {
     beforeAll(() => {
-      wrapper = render({ user });
+      wrapper = render({ user: { roles: [ROLES.CHECKER] } });
     });
 
     itRendersAHomeLink();
     itRendersAReportsLink();
+
+    itDoesNotRenderAUtilisationReportUploadLink();
+    itDoesNotRenderAPreviousReportsLink();
     itDoesNotRenderAUsersLink();
   });
 
-  describe.each(rolesToDisplayHomeNavigationItem)('viewed by a %s', (role) => {
-    const user = { roles: [role] };
+  describe(`viewed by role '${ROLES.ADMIN}'`, () => {
+    beforeAll(() => {
+      wrapper = render({ user: { roles: [ROLES.ADMIN] } });
+    });
 
+    itRendersAHomeLink();
+    itRendersAReportsLink();
+    itRendersAUsersLink();
+
+    itDoesNotRenderAUtilisationReportUploadLink();
+    itDoesNotRenderAPreviousReportsLink();
+  });
+
+  describe(`viewed by role '${ROLES.READ_ONLY}'`, () => {
     beforeAll(() => {
-      wrapper = render({ user });
+      wrapper = render({ user: { roles: [ROLES.READ_ONLY] } });
     });
 
     itRendersAHomeLink();
+
+    itDoesNotRenderAReportsLink();
+    itDoesNotRenderAUtilisationReportUploadLink();
+    itDoesNotRenderAPreviousReportsLink();
+    itDoesNotRenderAUsersLink();
+  });
+
+  describe(`viewed by role '${ROLES.PAYMENT_REPORT_OFFICER}'`, () => {
+    beforeAll(() => {
+      wrapper = render({ user: { roles: [ROLES.PAYMENT_REPORT_OFFICER] } });
+    });
+
+    itRendersAUtilisationReportUploadLink();
+    itRendersAPreviousReportsLink();
+
+    itDoesNotRenderAHomeLink();
+    itDoesNotRenderAReportsLink();
+    itDoesNotRenderAUsersLink();
+  });
+
+  describe('viewed with no roles', () => {
+    beforeAll(() => {
+      wrapper = render({ user: { roles: [] } });
+    });
+
+    itDoesNotRenderAHomeLink();
```

Comment:
In previous behaviour a user would have a home link only, while I'm not sure if this is right or wrong I'd prefer to not change existing behaviour so can we keep that logic please?